src/test/java/org/mule/service/http/impl/functional/client/HttpClientTlsCustomProviderTestCase.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.http.impl.functional.client;
+
+import static org.mule.runtime.api.metadata.MediaType.JSON;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.OK;
+import static org.mule.runtime.http.api.HttpConstants.Method.POST;
+import static org.mule.runtime.http.api.HttpConstants.Protocol.HTTP;
+import static org.mule.runtime.http.api.HttpConstants.Protocol.HTTPS;
+import static org.mule.service.http.impl.AllureConstants.HttpFeature.HttpStory.TLS;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.security.Security.insertProviderAt;
+import static java.security.Security.removeProvider;
+import static java.util.Collections.singletonList;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Rule;
+import org.mule.rules.BouncyCastleProviderCleaner;
+import org.mule.runtime.api.lifecycle.CreateException;
+import org.mule.runtime.api.tls.TlsContextFactory;
+import org.mule.runtime.core.api.util.IOUtils;
+import org.mule.runtime.http.api.client.HttpClient;
+import org.mule.runtime.http.api.client.HttpClientConfiguration;
+import org.mule.runtime.http.api.domain.entity.ByteArrayHttpEntity;
+import org.mule.runtime.http.api.domain.entity.multipart.HttpPart;
+import org.mule.runtime.http.api.domain.entity.multipart.MultipartHttpEntity;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+import org.mule.runtime.http.api.domain.message.response.HttpResponseBuilder;
+import org.mule.runtime.http.api.server.HttpServerConfiguration;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Story;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jsse.provider.BouncyCastleJsseProvider;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import java.io.IOException;
+import java.util.Collection;
+
+@Story(TLS)
```

Comment:
Add @Feature and @Issue Allure annotations

Reply:
Added @issue but @feature was defined at superclass